src/services/user.service.js

```diff
@@ -0,0 +1,116 @@
+const { ApiError } = require('../exceptions/ApiError.js');
+const { User } = require('../models/User.js');
+const { emailService } = require('./email.service.js');
+const bcrypt = require('bcrypt');
+const { v4: uuidv4 } = require('uuid');
+
+function getAllActive() {
+  return User.findAll({ where: { activationToken: null }, order: ['id'] });
+}
+
+function getByEmail(email) {
+  return User.findOne({ where: { email } });
+}
+
+function normalize({ id, email }) {
+  return { id, email };
+}
+
+async function register({ email, password }) {
+  const existingUser = await getByEmail(email);
+
+  if (existingUser) {
+    throw ApiError.BadRequest('Validation error', {
+      email: 'Email is already taken',
+    });
+  }
+
+  const activationToken = uuidv4();
+  const hash = await bcrypt.hash(password, 10);
+
+  await User.create({
+    email,
+    password: hash,
+    activationToken,
+  });
+
+  await emailService.sendActivationLink(email, activationToken);
+}
+
+async function resetEmail(email) {
+  const resetToken = uuidv4();
+  const user = await User.findOne({ where: { email } });
+
+  if (!user) {
+    throw ApiError.BadRequest('User not found');
+  }
+
+  await emailService.sendReset(email, resetToken);
+
+  user.resetToken = resetToken;
+  await user.save();
+}
+
+async function changePassword(userName, newPassword) {
+  const user = await User.findOne({ where: { userName } });
+
+  if (!user) {
+    throw ApiError.BadRequest('User not found');
+  }
+
+  const hash = await bcrypt.hash(newPassword, 10);
+
+  user.password = hash;
+  user.resetToken = null;
+
+  await user.save();
+}
+
+async function changeName(userName, newName) {
+  const user = await User.findOne({ where: { userName } });
+  const existUser = await User.findOne({ where: { userName: newName } });
+
+  if (existUser) {
+    throw ApiError.BadRequest('Username is already taken');
+  }
+
+  if (!user) {
+    throw ApiError.BadRequest('User not found');
```

Comment:
Consider checking if the current user exists before checking if the new username is already taken. This will improve the clarity and efficiency of the `changeName` function.